### ISSUE TITLE
Switch to Tesla as HTTP client

### ISSUE THREAD

**Issue by alexgleason** [seated/apus#114]
In the spirit of adapter-based clients, it would be good if the http client were configurable, too.

[Tesla](https://github.com/teamon/tesla) is the best Elixir HTTP client because it's all of them and none of them.

**Comment by swelham** [comment]
I agree that having the http client configurable would be a great enhancement. However, I would first look to consider an approach that doesn't require a new dependency.

Open to any suggestions on this.
